sdk/storage/storage-blob/samples/javascript/basic.js

```diff
@@ -88,12 +88,12 @@ async function main() {
 // A helper method used to read a Node.js readable stream into string
 async function streamToString(readableStream) {
   return new Promise((resolve, reject) => {
-    const chunks = [];
+    let buffer;
     readableStream.on("data", (data) => {
-      chunks.push(data.toString());
+      buffer = buffer ? Buffer.concat([buffer, data]) : data;
```

Comment:
According to the [documentation](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_concat_list_totallength), `Buffer.concat(list[, totalLength])` only works for Uint8Array and Buffer list.

What if the `data` is a string? Better to convert `data` to Buffer first if it's a string.

Reply:
I've tried to download a few file types. txt/json, pdf, jpg, png and mp4. They all seem to be returned as Buffers. 

Sadly, not all files seem to work. png, pdf and mp4 don't work. I'm going to use an http download library instead of this library.

Reply:
Sorry, I've responded to soon. It does not work properly when writing the buffer with fs to the disk. If I push the buffer through node.js express send(), then it works properly.